bgpd/bgp_fsm.c

```diff
@@ -180,6 +180,15 @@ static struct peer *peer_xfer_conn(struct peer *from_peer)
 	 * otherwise we'll get spurious failures during session establishment.
 	 */
 	frr_with_mutex(&peer->io_mtx, &from_peer->io_mtx) {
+		/*
+		 * Before exchanging FD remove doppelgnger from
+		 * keepalive peer hash, if config node peer fd
+		 * is set to -1. If blocked on lock then keepalive
+		 * thread can access doppelgnger pointer with fd -1.
```

Comment:
I don't understand this comment, can you elaborate a bit more? In what scenario would we have keepalives turned on but the peer fd set to -1?